### DISCUSSION TITLE
F-15C new keyfile support

### DISCUSSION THREAD

**Discussion by chihirobelmo** [chihirobelmo/FalconBMS-Alternative-Launcher#108] [General]
4.37.3 re-load key file everytime user enter 3D.
If user selects BMS - FULL.key it reads it if you fly F-16 but reads BMS - FULL-F15ABCD.key instead if its F-15C you fly. 
so its <filename><prefix>.key where F-16 does not have prefix but F-15 does. and future new A/C will have other prefixes.

**Comment by chihirobelmo** [comment]
`DeviceControl` class handles whole joystick assign state like whick loystick which button assins which callback. I think we now have to have 2 instances of DeviceControl for F-16 and F-15 and create 2 unique key files for each aircraft.

**Reply by MattCFII** [reply]
Just thinking and maybe too late for this next release, but can/did you do a different axis mapping too? I have a side mount X65 force sensitive stick for F-16 and then a center traditional stick for non-Viper planes.

**Comment by chihirobelmo** [comment]
We also have to hold unique instances for `keyFile` class.

**Comment by chihirobelmo** [comment]
for UI I'd like to put a dropdown box to select F-16 or F-15 assign active for edit.

**Reply by chihirobelmo** [reply]
might work like DCS UI i'd say

**Comment by arithex** [comment]
https://github.com/chihirobelmo/FalconBMS-Alternative-Launcher/pull/109

**Comment by arithex** [comment]
Shipped in v2.4.0.6 (in BMS 4.37.3)
